### PR TITLE
Add analyzer/fixer for ensuring no blank lines between close braces in C#

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
roslyn fix all changes here: https://github.com/dotnet/roslyn/pull/43699

### CODE REVIEW

**Comment by Evangelink** [comment]
It feels weird to have such analyzer in this repo. I would expect such analyzer to be either on StyleCop analyzer repo or to be an `IDEXXXX` so Roslyn repo.

**Comment by mavasani** [comment]
Following up with @CyrusNajmabadi offline..

**Comment by CyrusNajmabadi** [comment]
@Evangelink this is specifically about enforcing Roslyn's defacto codestyle within itself. It's not intended to be a general style for other projects.

**Comment by Evangelink** [comment]
@CyrusNajmabadi Ok and I don't want to block you or anything but you basically did most of the job to have something available for others too.

**Comment by CyrusNajmabadi** [comment]
I'm ok with some process taking this work and elevating it to a location where it can be used by all.  It's just out of scope for my particular goals, and i don't want to block forward progress on htat with an optional externality :)

File: src/Roslyn.Diagnostics.Analyzers/CSharp/AnalyzerReleases.Unshipped.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ### New Rules
 Rule ID | Category | Severity | Notes
 --------|----------|----------|-------
-RS0038 | RoslynDiagnosticsMaintainability | Warning | PreferNullLiteral
\ No newline at end of file
+RS0038 | RoslynDiagnosticsMaintainability | Warning | PreferNullLiteral
+RS0046 | RoslynDiagnosticsMaintainability | Warning | CSharpBracePlacementDiagnosticAnalyzer
```
Comment (mavasani) [inline src/Roslyn.Diagnostics.Analyzers/CSharp/AnalyzerReleases.Unshipped.md:5]:
```suggestion
RS0102 | RoslynDiagnosticsMaintainability | Warning | CSharpBracePlacementDiagnosticAnalyzer
```

File: src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticIds.cs [hunk 2]
```diff
@@ -49,5 +49,7 @@ internal static class RoslynDiagnosticIds
         public const string DoNotCallGetTestAccessorRuleId = "RS0043";
         public const string CreateTestAccessorRuleId = "RS0044";
         public const string ExposeMemberForTestingRuleId = "RS0045";
+
+        public const string BracePlacementRuleId = "RS0046";
```
Comment (mavasani) [inline src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticIds.cs:53]:
```suggestion
        public const string BracePlacementRuleId = "RS0102";
```

**Pull request merged**